app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :author, class_name: 'User'
+
+  def update_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] **[OPTIONAL]** Just you have done in the Comment model. It would be nice to make this method private and then add an `after_save` callback for this method. example

  ```
  after_save :update_counter

  private

  def update_counter
    author.increment!(:posts_counter)
  end
  ```

  Now every time a post is saved this method will automatically increment the author post_counter, without you calling it yourself